### ISSUE TITLE
Broken OIDC authentication flow

### ISSUE THREAD

**Issue by vishal423** [jhipster/generator-jhipster-quarkus#248]
**Describe the bug**

OIDC authentication flow seems broken as the application fails to redirect to Keycloak/OIDC providers to authenticate

**To Reproduce**
Generate application with OIDC authentication type. Start services and on clicking login, you shall see a JSON response on browser

**Expected behavior**
Should redirect to OIDC providers to authenticate

**Screenshots**

**Desktop (please complete the following information):**

-   OS: MacOS
-   Browser Safari

**Smartphone (please complete the following information):**
N/A

**Additional context**
It seems the issue got introduced after problem library integration.

**Comment by mraible** [comment]
Do you have a link to the Quarkus problem library?

**Comment by mraible** [comment]
I was able to use the following code to change the `oauth2/authorization/oidc` request's 401 response from JSON. However, it still doesn't redirect to Keycloak. 

```java
package com.company.myapp.web.rest.errors;

import io.quarkus.security.UnauthorizedException;

import javax.annotation.Priority;
import javax.ws.rs.core.Response;
import javax.ws.rs.ext.ExceptionMapper;
import javax.ws.rs.ext.Provider;

@Provider
@Priority(1)
public class UnauthorizedExceptionMapper implements ExceptionMapper<UnauthorizedException> {

    @Override
    public Response toResponse(UnauthorizedException exception) {
        return Response.status(Response.Status.UNAUTHORIZED).build();
    }
}
```

**Comment by mraible** [comment]
FYI... @ebullient

**Comment by lwitkowski** [comment]
@vishal423 I'll have a look today, one option is missing `WWW-Authenticate` header but I have to investigate it first to draw any conclusions :)

**Comment by lwitkowski** [comment]
@vishal423 would you mind bumping `resteasy-problem.version` to 2.1.0 which I've just released and synced with maven central? I've tested it on your jhipster blueprint branch and login functionality works fine.

The problem I see there though is missing `GET /api/account` endpoint called from frontend, but I don't think it has anything to do with problem extension.

**Comment by mraible** [comment]
@lwitkowski I can confirm that upgrading to RESTEasy Problem 2.1.0 solves the problem. Thank you!

**Comment by vishal423** [comment]
Thanks for quick turn around. I see Matt has already verified and bumped the version. I am traveling and wouldn't have access to laptop before mid November to test.

**Comment by mraible** [comment]
Closing because this is fixed in #247.
